### DISCUSSION TITLE
`cn` is redundant if there's only Tailwind Variants in its callsite

### DISCUSSION THREAD

**Discussion by armchair-traveller** [huntabyte/shadcn-svelte#615] [General]
Tailwind Variants uses Tailwind merge under-the-hood for classes passed into it. 

According to their home page:
> Forget about TailwindCSS style conflicts. Tailwind Variants efficiently merges conflicting styles.

From [first page of docs](https://www.tailwind-variants.org/docs/introduction):

> Automatic conflict resolution
> Tailwind Variants implements tailwind-merge under the hood, so it will efficiently merge your classes, so you don't have to worry about TailwindCSS class conflicts.

so when we have something like

```svelte
<ButtonPrimitive.Root
	{builders}
	class={cn(buttonVariants({ variant, size, className }))} <==== `cn` here has no purpose, only 1 class string used (className)
	type="button"
	{...$$restProps}
	on:click
	on:keydown
>
	<slot />
</ButtonPrimitive.Root>
```

`cn` actually runs the same code unnecessarily (clsx + tailwind-merge).

[Docs for overriding styles with class/className](https://www.tailwind-variants.org/docs/overriding-styles)

Anyway it's a small issue and almost a nitpick. I figured I'd mention I noticed it, I'd write an AST script but don't really have time to see which directories should be examined to affect the generated code.

**Comment by Greenheart** [comment]
I also realized the features I needed from `cn()` were already handled by `tailwind-variants`. I ended up removing `cn()` and the top-level dependencies `tailwind-merge` and `clsx`.

However, as noted in https://github.com/huntabyte/shadcn-svelte/issues/762, if you need the object syntax from `clsx`, you might want to keep using `clsx` at least.
